logstash-core/src/test/java/org/logstash/EventTest.java

```diff
@@ -193,6 +193,26 @@ public void testGetFieldList() throws Exception {
         assertEquals(1L, e.getField("[foo][0]"));
     }
 
+
+    @Test
+    public void testGetFieldDefaultValue() throws Exception {
+        Event e = new Event();
+
+        assertNull(e.getField("foo"));
+        assertEquals(1, e.getField("foo", 1));
+        e.setField("foo", 42L);
+        assertEquals(42L, e.getField("foo"));
+
+        Map<String, Object> inner = new HashMap<>(2);
+        inner.put("innerFoo", 42L);
+
+        e.setField("bar", inner);
```

Comment:
This is probably not an expected use of the event API (setting a value to a HashMap, I mean).

Can you rewrite this to be:

```
e.setField("[bar][innerFoo]", 42L);
```